### PR TITLE
test(dm): add a test to make sure there's no problem :P

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: ref #7493 

### What is changed and how it works?

simply add a test, no code change

#### Tests 

 - Unit test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: dm/syncer/syncer_test.go [hunk 1]
```diff
@@ -1676,111 +1677,103 @@ func (s *testSyncerSuite) TestTrackDownstreamTableWontOverwrite(c *C) {
...
+	syncer.downstreamTrackConn = dbconn.NewDBConn(cfg, conn.NewBaseConn(dbConn, &retry.FiniteRetryStrategy{}))
 
-	c.Assert(ti, DeepEquals, ti2)
+	for _, c := range cases {
+		syncer.schemaTracker, err = schema.NewTestTracker(ctx, cfg.Name, syncer.downstreamTrackConn, log.L())
+		require.NoError(t, err)
+
+		mock.ExpectQuery("SHOW VARIABLES LIKE 'sql_mode'").WillReturnRows(
+			sqlmock.NewRows([]string{"Variable_name", "Value"}).AddRow("sql_mode", ""))
+		// create table t (c bigint primary key auto_random);
```
Comment (lance6716) [inline dm/syncer/syncer_test.go:1748]:
```suggestion
```

File: dm/syncer/syncer_test.go [hunk 2]
```diff
@@ -1676,111 +1677,102 @@ func (s *testSyncerSuite) TestTrackDownstreamTableWontOverwrite(c *C) {
...
+				") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin /*T![placement] PLACEMENT POLICY=`on_ssd` */",
+			"create table tbl (c bigint primary key) /*T![placement] PLACEMENT POLICY=`on_ssd` */",
+		},
+	}
 
-	ti.ID = ti2.ID
-	ti.UpdateTS = ti2.UpdateTS
+	db, mock, err := sqlmock.New()
+	require.NoError(t, err)
+	require.NoError(t, err)
```
Comment (okJiang) [inline dm/syncer/syncer_test.go:1735]:
```suggestion
	require.NoError(t, err)
```

duplicate

**Review by okJiang — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**